lib/foreman/renderer/configuration.rb

```diff
@@ -19,6 +19,8 @@ class Configuration
         :medium_provider,
         :medium_uri,
         :load_hosts,
+        :load_users,
+        :user_auth_source,
```

Comment:
where do we use these? Are they needed?

Reply:
These methods are necessary for correct working of https://github.com/theforeman/community-templates/pull/660. So yes, they're needed.

Reply:
:+1: yeah, I haven't seen these used, but make sense as this is the same issue :)
I am still not sure about the usefullnes of `user_auth_source`, could be called directly in the template, right? But it's ok, I guess :)

Reply:
The main benefit is, that it handles the case, where user's auth source is nil for whatever reason. I don't want users to write such logic in templates. Also ideally they shouldn't need to know the internal structure of AR models.

But I'd suggest one change here. It's not clear from the method name it returns the name of the auth source. I'd expect to get AuthSource instance. Can we call it `user_auth_source_name` please?